src/Definition/Reference.php

```diff
@@ -13,15 +15,25 @@
  */
 class Reference implements Definition, SelfResolvingDefinition
 {
+    public static $serviceLocatorClass = ServiceLocator::class;
+
     /** Entry name. */
     private string $name = '';
 
-    /**
-     * @param string $targetEntryName Name of the target entry
-     */
+    private bool $isServiceLocatorEntry;
+
     public function __construct(
+        /**
+         * @var string Name of the target entry
+         */
         private string $targetEntryName,
+        /**
+         * @var string|null name of an entry - holder of a definition requesting this entry
+         */
+        private ?string $requestingName = null,
+        private ?ServiceLocatorDefinition $serviceLocatorDefinition = null
```

Comment:
please remove this from constructor parameters, this should not be altered from outside. Ref: https://github.com/ovos/PHP-DI/pull/7/files#diff-addce6eaa59f011f6a80f000d9b021f007d8410be8044e3a747a6aa93ac54053R55